data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/IndexManager.java

```diff
@@ -34,14 +43,98 @@
     protected RestHighLevelClient restHighLevelClient;
     protected OpenSearchSinkConfiguration openSearchSinkConfiguration;
     protected IsmPolicyManagementStrategy ismPolicyManagementStrategy;
+    protected String indexPrefix;
 
     private static final Logger LOG = LoggerFactory.getLogger(IndexManager.class);
 
+    //For matching a string that begins with a "%{" and ends with a "}".
+    //For a string like "data-prepper-%{yyyy-MM-dd}", "%{yyyy-MM-dd}" is matched.
+    private final static String TIME_PATTERN_REGULAR_EXPRESSION  = "%\\{.*?\\}";
+
+    //For matching a string enclosed by "%{" and "}".
+    //For a string like "data-prepper-%{yyyy-MM}", "yyyy-MM" is matched.
+    private final static String TIME_PATTERN_INTERNAL_EXTRACTOR_REGULAR_EXPRESSION  = "%\\{(.*?)\\}";
+
+    private Optional<DateTimeFormatter> indexTimeSuffixFormatter;
+    private final static ZoneId UTC_ZONE_ID = ZoneId.of(TimeZone.getTimeZone("UTC").getID());
+
     protected IndexManager(final RestHighLevelClient restHighLevelClient, final OpenSearchSinkConfiguration openSearchSinkConfiguration){
         checkNotNull(restHighLevelClient);
         checkNotNull(openSearchSinkConfiguration);
         this.restHighLevelClient = restHighLevelClient;
         this.openSearchSinkConfiguration = openSearchSinkConfiguration;
+        initializeIndexPrefixAndSuffix();
+    }
+
+    private void initializeIndexPrefixAndSuffix(){
+        String indexAliasFromConfig = openSearchSinkConfiguration.getIndexConfiguration().getIndexAlias();
+        indexPrefix = indexAliasFromConfig.replaceAll(TIME_PATTERN_REGULAR_EXPRESSION, "");
+
+        final Pattern pattern = Pattern.compile(TIME_PATTERN_INTERNAL_EXTRACTOR_REGULAR_EXPRESSION);
+        final Matcher matcher = pattern.matcher(indexAliasFromConfig);
+        if (matcher.find()) {
+            String timePattern = matcher.group(1);
```

Comment:
What happens if someone provides 2 patterns like this: `data-prepper-%{yyyy}-%{MM}`. I think we would throw an error that indicates "Time pattern can only be a suffix of an index". Do we support 2 or more patterns? Can we throw a specific error that is more helpful to the user?

Reply:
`final String timePattern = matcher.group(1);`

Reply:
Good point, added new check: 
            if (matcher.groupCount() > 1) {
                throw new IllegalArgumentException("Only support one date-time pattern.");
            }